brainglobe_atlasapi/atlas_generation/atlas_scripts/demba_dev_mouse.py

```diff
@@ -0,0 +1,345 @@
+    meshes_dir_path.mkdir(exist_ok=True)
+    tree = get_structures_tree(structures)
+    labels = np.unique(annotated_volume).astype(np.int32)
+    for key, node in tree.nodes.items():
+        if key in labels:
+            is_label = True
+        else:
+            is_label = False
+
+        node.data = Region(is_label)
+
+    # Mesh creation
+    closing_n_iters = 2  # not used for this atlas
+    decimate_fraction = 0.2  # not used for this atlas
+    smooth = False
+    for node in track(
+        tree.nodes.values(),
+        total=tree.size(),
+        description="Creating meshes",
+    ):
+        create_region_mesh(
+            (
+                meshes_dir_path,
+                node,
+                tree,
+                labels,
+                annotated_volume,
+                METADATA.root_id,
+                closing_n_iters,
+                decimate_fraction,
+                smooth,
+            )
+        )
+    # Create meshes dict
+    meshes_dict = dict()
+    structures_with_mesh = []
+    for s in structures:
+        # Check if a mesh was created
+        mesh_path = meshes_dir_path / f'{s["id"]}.obj'
+        if not mesh_path.exists():
+            print(f"No mesh file exists for: {s}, ignoring it")
+            continue
+        else:
+            # Check that the mesh actually exists (i.e. not empty)
+            if mesh_path.stat().st_size < 512:
+                print(f"obj file for {s} is too small, ignoring it.")
+                continue
+
+        structures_with_mesh.append(s)
+        meshes_dict[s["id"]] = mesh_path
+
+    print(
+        f"In the end, {len(structures_with_mesh)} "
+        "structures with mesh are kept"
+    )
+    return meshes_dict
+
+
+### If the code above this line has been filled correctly, nothing needs to be
+### edited below (unless variables need to be passed between the functions).
+if __name__ == "__main__":
+    bg_root_dir = Path.home() / "brainglobe_workingdir" / METADATA.name
+    bg_root_dir.mkdir(exist_ok=True)
+    download_resources(
+        download_dir_path=bg_root_dir,
+        atlas_name=METADATA.name,
+        atlas_file_url=data_file_url,
+    )
+    meshes_dict = None
+
+    for age in range(4, 57):
+        age_specific_root_dir = bg_root_dir / f"P{age}"
+        age_specific_root_dir.mkdir(exist_ok=True)
+        for resolution, modalities in resolution_to_modalities.items():
+            reference_volume, annotated_volume = (
+                retrieve_reference_and_annotation(
+                    bg_root_dir, age, resolution, modalities[0]
+                )
+            )
+            if len(modalities) > 1:
+                additional_references = retrieve_additional_references(
+                    bg_root_dir, age, resolution, modalities[1:]
+                )
+            hemispheres_stack = retrieve_hemisphere_map()
+            structures = retrieve_structure_information()
+            if meshes_dict is None:
```

Comment:
```suggestion
           # generate pixel-scale mesh files only once, for 25um, and 
           # re-use them and the meshes_dict
            if meshes_dict is None:
```
Is what I think is happening?

Reply:
correct!